test/show.jl

```diff
@@ -1,19 +1,103 @@
 module TestShow
     using Base.Test, DataFrames
 
-    df = DataFrame(A = 1:3, B = ["x", "y", "z"])
+    # In the future newline characte \n should be added to this test case
+    df = DataFrame(A = 1:4, B = ["x\"", "∀ε⫺0: x+ε⫺x", "z\$", "ABC"],
+                   C = Float32[1.0, 2.0, 3.0, 4.0])
+    srand(1)
+    df_big = DataFrame(rand(50,50))
+
+    refstr = """
+    4×3 DataFrames.DataFrame
+    │ Row │ A │ B             │ C   │
+    ├─────┼───┼───────────────┼─────┤
+    │ 1   │ 1 │ x\"            │ 1.0 │
+    │ 2   │ 2 │ ∀ε⫺0: x+ε⫺x │ 2.0 │
+    │ 3   │ 3 │ z\$            │ 3.0 │
+    │ 4   │ 4 │ ABC           │ 4.0 │"""
+
+    for f in [show, showall], allcols in [true, false]
+        io = IOBuffer()
+        f(io, df, allcols)
+        str = String(take!(io))
+        @test str == refstr
+    end
+
+    refstr = """
+    4×3 DataFrames.DataFrame
+    
+    │ Col # │ Name │ Eltype  │ Missing │
+    ├───────┼──────┼─────────┼─────────┤
+    │ 1     │ A    │ Int64   │ 0       │
+    │ 2     │ B    │ String  │ 0       │
+    │ 3     │ C    │ Float32 │ 0       │"""
+    for a in [true, false]
+        io = IOBuffer()
+        showcols(io, df, a, false)
+        str = String(take!(io))
+        @test str == refstr
+    end
+
+    refstr = """
+    4×3 DataFrames.DataFrame
+    
+    │ Col # │ Name │ Eltype  │ Missing │ Values          │
+    ├───────┼──────┼─────────┼─────────┼─────────────────┤
+    │ 1     │ A    │ Int64   │ 0       │ 1  …  4         │
+    │ 2     │ B    │ String  │ 0       │ \"x\\\"\"  …  \"ABC\" │
+    │ 3     │ C    │ Float32 │ 0       │ 1.0  …  4.0     │"""
+    for a in [true, false]
+        io = IOBuffer()
+        showcols(io, df, a, true)
+        str = String(take!(io))
+        @test str == refstr
+    end
 
     io = IOBuffer()
-    show(io, df)
-    show(io, df, true)
-    showall(io, df)
-    showall(io, df, true)
+    show(io, df_big)
```

Comment:
Could you also test the output of these functions (even if that's verbose, it's fine)? Else the case when there are too many rows won't be covered. You should probably pass a custom `IOContext` to control the size of the display. Also better define `df_big` here rather than above, where it isn't used.